dig.go

```diff
@@ -202,15 +202,45 @@ func (c *Container) isAcyclic(n node) error {
 	return detectCycles(n, c.nodes, nil)
 }
 
-// Retrieve a type from the container
-func (c *Container) get(t reflect.Type) (reflect.Value, error) {
+type keyOpt func(*key)
```

Comment:
can technically specify `getOpt` and later `setOpt`, but they will both operate on keys and look somewhat similar.